Fix \Knp\Menu\MenuItem @Label issue

VincentLanglet opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 3.x is for everything backwards compatible, like patches, features and deprecation notices
    - master is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/CONTRIBUTING.md#the-base-branch
-->
I am targeting this branch, because it is a bug fix.

Closes #6031

## Changelog

<!-- MANDATORY 
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
    This will end up on https://github.com/sonata-project/SonataAdminBundle/releases,
    please keep it short and clear and to the point
-->

<!-- 
    If you are updating something that doesn't require
    a release, you can delete the whole Changelog section.
    (eg. update to docs, tests)
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown

### Fixed
- Call \Knp\Menu\MenuItem::getLabel() method directly in twig template to avoid a possible side effect from \ArrayAccess.
```

<!--
    If this is a work in progress, uncomment this section.
    You can add as many tasks as you want.
    If some are not relevant, just remove them.
    
    ## To do
    
    - [ ] Update the tests
    - [ ] Update the documentation
    - [ ] Add an upgrade note
-->


## Description:

\Knp\Menu\MenuItem in knplabs/knp-menu-bundle implements \ArrayAccess and defines its offsetGet method as follows:
```php
    /**
     * Implements ArrayAccess
     */
    public function offsetGet($name)
    {
        return $this->getChild($name);
    }

    public function getChild($name)
    {
        return isset($this->children[$name]) ? $this->children[$name] : null;
    }

```

It has a property `$label` with its getter: 
```php
    /**
     * Label to output, name is used by default
     *
     * @var string|null
     */
    protected $label;

    public function getLabel()
    {
        return (null !== $this->label) ? $this->label : $this->name;
    }
```
## Problem :

If a MenuItem object has a child with the key `label`, the twig expression `menu.label` will trigger the `offsetGet` method of the \ArrayAccess and return the child instead of the `$label` property of the MenuItem object.

This would result in Catchable Fatal Error: Object of class Knp\Menu\MenuItem could not be converted to string.

## Solution:

Call the getter of the `$label` property directly:
```twig
menu.getLabel()
```

jordisala1991 commented:
Thank you @VincentLanglet

franmomu commented:
IMHO it would be nice to add the functional tests added in the original PR https://github.com/sonata-project/SonataAdminBundle/pull/6031/files

VincentLanglet commented:
> IMHO it would be nice to add the functional tests added in the original PR https://github.com/sonata-project/SonataAdminBundle/pull/6031/files

I added one, without creating a whole class/admin called PR6301.
The `configureTabMenu` I added was failing without this PR. And doesn't fail now. Isn't that enough ?

franmomu commented:
Well, the current test do not test twig calls, https://github.com/sonata-project/SonataAdminBundle/pull/6031#pullrequestreview-391862126. In fact, when the test was added, [it found another place where was wrong](https://github.com/sonata-project/SonataAdminBundle/pull/6031#issuecomment-612662697).
